src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt

```diff
@@ -0,0 +1,150 @@
+package com.safehill.kclient.tasks.syncing
+
+import com.safehill.kclient.models.dtos.ConversationThreadAssetDTO
+import com.safehill.kclient.models.dtos.ConversationThreadOutputDTO
+import com.safehill.kclient.models.dtos.InteractionsThreadSummaryDTO
+import com.safehill.kclient.models.dtos.MessageOutputDTO
+import com.safehill.kclient.models.dtos.websockets.ConnectionAck
+import com.safehill.kclient.models.dtos.websockets.TextMessage
+import com.safehill.kclient.models.dtos.websockets.ThreadAssets
+import com.safehill.kclient.models.dtos.websockets.ThreadCreated
+import com.safehill.kclient.models.dtos.websockets.UnknownMessage
+import com.safehill.kclient.models.dtos.websockets.WebSocketMessage
+import com.safehill.kclient.models.interactions.InteractionAnchor
+import com.safehill.kclient.network.ServerProxy
+import com.safehill.kclient.network.WebSocketApi
+import com.safehill.kclient.tasks.BackgroundTask
+import com.safehill.kclient.util.runCatchingPreservingCancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+
+class ThreadInteractionSync(
+    private val serverProxy: ServerProxy,
+    private val threadInteractionSyncListener: InteractionSyncListener,
+    private val webSocketApi: WebSocketApi
+) : BackgroundTask {
+
+    override suspend fun run() {
+        coroutineScope {
+            syncThreadInteractionSummary()
+            webSocketApi.socketMessages.collect {
+                launch { it.handle() }
+            }
+        }
+    }
+
+    private suspend fun WebSocketMessage.handle() {
+        when (this) {
+            is TextMessage -> {
+                this.notifyTextMessage()
+            }
+
+            is ThreadAssets -> {
+                this.notifyNewAssetInThread()
+            }
+
+            is ThreadCreated -> {
+                this.notifyCreationOfThread()
+            }
+
+            UnknownMessage, is ConnectionAck -> {}
+        }
+    }
+
+    private suspend fun ThreadCreated.notifyCreationOfThread() {
+        threadInteractionSyncListener.didAddThread(
+            threadDTO = thread
+        ).also {
+            serverProxy.localServer.createOrUpdateThread(
+                threads = listOf(thread)
+            )
+        }
+    }
+
+    private suspend fun ThreadAssets.notifyNewAssetInThread() {
```

Comment:
Is this for both photo messages and non?

Reply:
No.. its only for photo messages.

Reply:
Non photo messages are given by RemoteDownloadOperation for now.